local_db/data_access.py

```diff
@@ -120,3 +143,37 @@ def add_file_to_request(
         # Return updated FileGroups data
         metadata = self._find_metadata(request_id)
         return self._get_filegroups(metadata)
+
+    def get_file_approvals(self, request_id):
+        return [
+            self._filereview(r)
+            for r in FileReview.objects.filter(
+                file__filegroup__request_id=request_id
+            ).all()
+        ]
+
+    def approve_file(self, request_id, user, relpath):
+        with transaction.atomic():
+            # nb. the business logic layer approve_file() should confirm that this path
+            # is part of the request before calling this method
+            request_file = RequestFileMetadata.objects.get(
+                filegroup__request_id=request_id, relpath=relpath
+            )
+
+            review, _ = FileReview.objects.get_or_create(
+                file=request_file, reviewer=user
+            )
+            review.status = FileApprovalStatus.APPROVED
+            review.save()
+
+    def reject_file(self, request_id, user, relpath):
+        with transaction.atomic():
+            request_file = RequestFileMetadata.objects.get(
+                filegroup__request_id=request_id, relpath=relpath
+            )
+
+            review, _ = FileReview.objects.get_or_create(
+                file=request_file, reviewer=user
+            )
+            review.status = FileApprovalStatus.REJECTED
```

Comment:
I think you also need to udpate `review.updated_at` for both these too?

Reply:
good spot, have set [`auto_now=True`](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.DateField.auto_now) instead

Reply:
oh! TIL about auto_now, nice. I bet REX would like to know about this!